lib/howl/ui/editor.moon

```diff
@@ -306,11 +306,12 @@ class Editor extends PropertyObject
     @buffer\as_one_undo -> f lines
 
   with_position_restored: (f) =>
-    line, column, indentation = @cursor.line, @cursor.column, @current_line.indentation
+    line, column, indentation, top_line = @cursor.line, @cursor.column, @current_line.indentation, @line_at_top
     status, ret = pcall f, self
     @cursor.line = line
     delta = @current_line.indentation - indentation
     @cursor.column = column + delta
+    @line_at_top = top_line
```

Comment:
Done but had to revert - didn't work as expected - i.e. doing that make the view 'jump' to another a slightly shifted view. Will this actually scroll twice, because the drawing will only happen once after it returns from Lua?